### PR TITLE
[INTEL MKL] Register matmul kernel to use Eigen based implementation for int64

### PR DESCRIPTION
**Pull request by bhavani-subramanian**
Recently, int64 has been enabled for matmul in TensorFlow (https://github.com/tensorflow/tensorflow/commit/2d0c8127d76b9a56c8e264cc98e902b257fb64b2). Since MKL does not support int64 for matmul, I have registered the matmul kernel to fallback to the default Eigen based implementation in such cases.

This PR would also cause `//tensorflow/python/kernel_tests:matmul_op_test` to pass with MKL-DNN.

### CODE REVIEW

**Comment by agramesh1** [comment]
Pinging @penpornk for a review.

File: tensorflow/core/kernels/matmul_op.cc [hunk 1]
```diff
@@ -580,13 +580,14 @@ struct MatMulFunctor<SYCLDevice, T> {
 
 #if defined(INTEL_MKL) && defined(ENABLE_MKL)
 
-// MKL does not support half, bfloat16 and int32 types for
+// MKL does not support half, bfloat16, int32 and int64 types for
```
Comment (penpornk) [inline tensorflow/core/kernels/matmul_op.cc]: For completeness, could you please add that MKL's `MatMul` kernels for `float`, `double`, `complex64`, and `complex128` are registered in `mkl_matmul_op.cc` before this sentence?

File: tensorflow/core/kernels/matmul_op.cc [hunk 2]
```diff
@@ -602,7 +603,7 @@ TF_CALL_float(REGISTER_CPU_EIGEN);
 TF_CALL_complex64(REGISTER_CPU);
 TF_CALL_complex128(REGISTER_CPU);
 TF_CALL_double(REGISTER_CPU);
-#else  // INTEL_MKL_DNN_ONLY
+#else   // INTEL_MKL_DNN_ONLY
```
Comment (penpornk) [inline tensorflow/core/kernels/matmul_op.cc]: The original code already has two spaces between `else` and `//`. Is there a reason for the third space?

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR! I have minor comments.

File: tensorflow/core/kernels/matmul_op.cc (hunk 2 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/matmul_op.cc]: @penpornk `clang-format` prompted me to add this additional space. I can remove it if you think it is superfluous.

Comment (penpornk) [reply tensorflow/core/kernels/matmul_op.cc]: Expanding the code, I understand why now. It's to align with <code>#endif &nbsp;// INTEL_MKL_DNN_ONLY</code> (Github didn't show that line in the changes). Please keep it.

File: tensorflow/core/kernels/matmul_op.cc (hunk 1 above)
Comment (bhavani-subramanian) [reply tensorflow/core/kernels/matmul_op.cc]: @penpornk Done.

**Review by penpornk — APPROVED** [review]
Thank you for the change!

File: tensorflow/core/kernels/matmul_op.cc (hunk 2 above)
Comment (penpornk) [reply tensorflow/core/kernels/matmul_op.cc]: This line is actually failing our `clang-format` check. Please remove the third space.
```suggestion
#else  // INTEL_MKL_DNN_ONLY
```

**Review by penpornk — CHANGES_REQUESTED** [review]

**Comment by bhavani-subramanian** [comment]
@penpornk I have removed the extra space per your suggestion.

**Review by penpornk — APPROVED** [review]
Thank you!

**Comment by bhavani-subramanian** [comment]
Hi @penpornk, I see that this PR is still not merged into the master. I was wondering if there's anything needed from my side to expedite this merge process. Thanks.

**Comment by penpornk** [comment]
@bhavani-subramanian Some of the checks failed. For example, GPU CC has 100 tests failed to build. Are you able to see the logs?

**Comment by bhavani-subramanian** [comment]
@penpornk Yes, I am able to see the logs. I'll look into these unit test failures. Thanks

**Comment by bhavani-subramanian** [comment]
@penpornk It looks like these failures are caused by internal commit to the master:  
image
As you can see, in `2c164ed`, all unit tests pass. However, starting from `fc6cd33`, XLA and GPU tests start to fail.
Am I missing something here?

**Comment by penpornk** [comment]
@bhavani-subramanian I'm not sure what is happening either. I don't think your changes would affect anything. I re-ran the tests again. Now GPU CC passed but Ubuntu CC that used to pass failed -- and it doesn't show the logs either. A recent commit (around the same time I re-ran the tests) passed all the tests. I also forced-run kokoro on this [typo PR](https://github.com/tensorflow/tensorflow/pull/23217) and it's passing all the completed tests so far. I'm thinking of pulling the PR in and see what happens directly.

@wt-huang Could you please help pull the PR in?

**Comment by wt-huang** [comment]
@penpornk just did, thanks.

**Pull request merged**